src/sqlfluff/rules/L043.py

```diff
@@ -240,10 +240,7 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                         description="Unnecessary CASE statement. "
                         "Use COALESCE function instead.",
                     )
-                elif (
-                    column_reference_segment.raw_segments_upper
-                    == then_expression.raw_segments_upper
-                ):
+                elif column_reference_segment.raw_upper == then_expression.raw_upper:
```

Comment:
Not sure why the original version used the other, more obscure property; `raw_upper` is more commonly used and works just fine.